services/oauth/v2/cp_connector.go

```diff
@@ -0,0 +1,145 @@
+package v2
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+type ControlPlaneConnectorI interface {
+	CpApiCall(cpReq *ControlPlaneRequestT) (int, string)
+}
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+type ControlPlaneConnector struct {
+	Client  HttpClient
+	Logger  logger.Logger
+	timeOut time.Duration
+}
+
+func NewControlPlaneConnector(options ...func(*ControlPlaneConnector)) ControlPlaneConnectorI {
+	cpConnector := &ControlPlaneConnector{}
```

Comment:
Let's also set some default values for timeout here?

Reply:
Driving this via a config `cpTimeoutDuration := config.GetDuration("HttpClient.oauth.timeout", 30, time.Second)` in `oauth.go`